pkg/data/components/ApplyK8sSemconvForLogs.yaml

```diff
@@ -0,0 +1,46 @@
+kind: ApplyOTelK8sSemconvProcessor
+name: Apply OpenTelemetry K8s Semantic Conventions
+style: processor
+type: base
+status: development
+version: v0.0.1
+summary: Applies OpenTelemetry Kubernetes semantic conventions 
+description: |-
+  This processor applies OpenTelemetry Kubernetes semantic conventions.
+  For log records, it includes setting the appropriate attributes for log severity, source, and other relevant metadata.
+tags:
+  - category:processor
+  - service:collector
+  - signal:OTelLogs
+ports:
+  # inputs
+  - name: Logs
+    direction: input
+    type: OTelLogs
+  # outputs
+  - name: Logs
+    direction: output
+    type: OTelLogs
+templates:
+  - kind: collector_config
+    name: otel_apply_k8s_semconv
+    format: collector
+    meta:
+      componentSection: processors
+      signalTypes: [logs]
+      collectorComponentName: transform
+    data:
+      - key: "{{ .ComponentName }}.error_mode"
+        value: "ignore"
+      - key: "{{ .ComponentName }}.log_statements[0]"
+        value: "merge_maps(attributes, attributes[\"metadata\"], \"upsert\") where IsMap(attributes[\"metadata\"])"
```

Comment:
This is another situation where the component is making an assumption that `attributes[\"metadata\"]` is gonna exist and have the interesting information we want. I think that is too big of an assumption to make.

Reply:
I think we should talk with Jessica again about if this specific of a component  is desired. The transform config these concepts came from is really specific to processing the k8sobjectreceiver's kubernetes logs. I dont think it would translate well to a generic solution for a customer.

Reply:
I've removed the map merge thing, which can be handled in another processor (see #216). Then updated this processor to specifically only apply otel semconv on k8s object receiver logs.

The other processors I've added recently are more generic, while this one is specific. Do you think that works better?

Reply:
I feel like this is hyper-specific to the k8sobjectreceiver's events. I've not seen these attributes be extracted in other scenarios as they are commonly added via the k8sattributesprocessor.